dev-libs/wlroots/wlroots-9999.ebuild

```diff
@@ -0,0 +1,118 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+if [[ ${PV} == 9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/swaywm/wlroots.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/swaywm/wlroots/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+inherit fcaps meson
+
+DESCRIPTION="Pluggable, composable, unopinionated modules for building a Wayland compositor"
+HOMEPAGE="https://github.com/swaywm/wlroots"
+LICENSE="MIT"
+SLOT="0"
+IUSE="elogind examples filecaps png rootston systemd x11-backend xcb-icccm xcb-xkb xwayland"
+REQUIRED_USE="?? ( elogind systemd )"
+
+# An example requires >=ffmpeg-4.0 (actually masked) or equivalent libav
+# (actually upstream is behind ffmpeg-4.0). So the ffmpeg use flag is actually
+# not present and the related example (only one) will not be build. This is a
+# reminder to add the 'ffmpeg' use flag when ffmpeg-4.0 will be unmasked.
+# examples? ( ffmpeg? ( >=media-video/ffmpeg-4.0 ) )
+RDEPEND=">=dev-libs/libinput-1.7.0:0=
+	>=dev-libs/wayland-1.16.0
+	>=dev-libs/wayland-protocols-1.15
+	media-libs/mesa[egl,gles2,gbm]
+	virtual/libudev
+	x11-libs/libdrm
+	x11-libs/libxkbcommon
+	x11-libs/pixman
+	filecaps? ( sys-libs/libcap )
+	elogind? ( >=sys-auth/elogind-237 )
+	examples? (
+		png? ( media-libs/libpng:* )
+	)
+	systemd? ( >=sys-apps/systemd-237 )
+	x11-backend? (
+		x11-libs/libxcb:0=
+		xcb-xkb? ( x11-libs/libxcb[xkb] )
+	)
+	xcb-icccm? ( x11-libs/xcb-util-wm )
+	xcb-xkb? ( x11-libs/libxcb[xkb] )
+	xwayland? (
+		x11-base/xorg-server[wayland]
+		x11-libs/libxcb:0=
+	)"
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig"
+
+FILECAPS=( cap_sys_admin usr/bin/rootston )
+
+src_configure() {
+	# xcb-util-errors is not on Gentoo Repository (and upstream seems inactive?)
+	local emesonargs=(
+		"-Dxcb-errors=disabled"
+		-Dlibcap=$(usex filecaps enabled disabled)
+		-Dxcb-icccm=$(usex xcb-icccm enabled disabled)
+		-Dxcb-xkb=$(usex xcb-xkb enabled disabled)
+		-Dxwayland=$(usex xwayland enabled disabled)
+		-Dx11-backend=$(usex x11-backend enabled disabled)
+		$(meson_use rootston)
+		$(meson_use examples)
+		"-Dwerror=false"
+	)
+	if use systemd ; then
+		emesonargs+=("-Dlogind=enabled" "-Dlogind-provider=systemd")
+	elif use elogind ; then
+		emesonargs+=("-Dlogind=enabled" "-Dlogind-provider=elogind")
+	else
+		emesonargs+=("-Dlogind=disabled")
+	fi
+
+	meson_src_configure
+}
+
+src_install() {
+	if use rootston ; then
+		dobin "${BUILD_DIR}"/rootston/rootston
+		newdoc rootston/rootston.ini.example rootston.ini
+		docompress -x usr/share/doc/"${PF}"/rootston.ini
+	fi
+
+	if use examples ; then
+		find "${BUILD_DIR}"/examples/* -type d -print0 | xargs -0 rm -rf || die
+		exeinto /usr/share/doc/"${PF}"/examples/
```

Comment:
The purpose of this directory is to store source code, not compiled programs.

Reply:
So, where should I place them? `/usr/bin`? Since it's not a good idea to store a `simple` or `idle` executable in `/usr/bin`, maybe I should prefix every executable with something like `wlroots-example-...`?

Reply:
Is there really a purpose in installing compiled examples? As I said, we usually support installing examples as sources that the user could read or copy to start writing his own programs. Here, I can't seem much gain from having them installed.

If you really think they're useful, then I guess `/usr/lib/wlroots` (or libexec) or something similar. However, you shouldn't be using USE=examples for that but some other flag to avoid confusion.

Reply:
I've removed them entirely.